src/domain/reference/reference.dto.ts

```diff
@@ -1,17 +1,19 @@
 import { InputType, Field } from '@nestjs/graphql';
-import { MaxLength } from 'class-validator';
+import { LONG_TEXT_LENGTH, MID_TEXT_LENGTH } from '@constants';
+import { IsDataURI, IsOptional, MaxLength } from 'class-validator';
 
 @InputType()
 export class ReferenceInput {
   @Field({ nullable: true })
-  @MaxLength(30)
+  @MaxLength(MID_TEXT_LENGTH)
   name!: string;
 
   @Field({ nullable: true })
-  @MaxLength(300)
+  @IsDataURI()
   uri!: string;
```

Comment:
This field contains URLs should be used `@IsUrl` instead.

Optionally if we want to store data in URI form we may need to find a way to apply two validators with `OR` operation. The default one is `AND`.